include/mbedtls/ssl.h

```diff
@@ -1935,27 +1963,50 @@ void mbedtls_ssl_conf_transport( mbedtls_ssl_config *conf, int transport );
 void mbedtls_ssl_conf_authmode( mbedtls_ssl_config *conf, int authmode );
 
 #if defined(MBEDTLS_SSL_PROTO_TLS1_3) && defined(MBEDTLS_SSL_EARLY_DATA)
+
 /**
-* \brief    Set the early data mode
-*           Default: disabled on server and client
-*
-* \param conf   The SSL configuration to use.
-* \param early_data_enabled can be:
-*
-*  MBEDTLS_SSL_EARLY_DATA_DISABLED:  early data functionality is disabled
-*                                    This is the default on client and server.
-*
-*  MBEDTLS_SSL_EARLY_DATA_ENABLED:  early data functionality is enabled and
-*                        may be negotiated in the handshake. Application using
-*                        early data functionality needs to be aware of the
-*                        lack of replay protection of the early data application
-*                        payloads.
-*
-* \warning This interface is experimental and may change without notice.
-*
-*/
+ * \brief    Set early data parameter.
+ *
+ * \param conf   The SSL configuration to use.
+ * \param max_early_data_size It is reused for client and server with different
+ *                            meaning.
+ *                            Server:
+ *                            The maximium amount of 0-RTT data in bytes on
+ *                            server. Enable/disable early data on client.
+ *                              - 0      Disable early_data.
+ *                              - others The maximum amount.
+ *                            Client: Enable or disable early data feature.
+ *                              - 0      Disable early_data
+ *                              - others Enable early_data.
+ *
+ * \warning This interface is experimental and may change without notice.
+ *
+ */
 void mbedtls_ssl_tls13_conf_early_data( mbedtls_ssl_config *conf,
-                                        int early_data_enabled );
+                                        size_t max_early_data_size );
+
+#if defined(MBEDTLS_SSL_SRV_C)
+/**
+ * \brief    Configure early data buffer management callback.
+ *
+ * \param conf   The SSL configuration to use.
+ * \param early_data_write_cb callback function for writing early data to buffer.
+ *                            - void *                : Context of callback
+ *                            It is a write like function.
+ * \param early_data_write_cb callback function for reading early data from buffer.
+ *                            - void *                : Context of callback
+ *                            It is a read like function.
+ * \param context             context of callback function.
+ *
+ * \warning This interface is experimental and may change without notice.
+ *
+ */
+void mbedtls_ssl_tls13_conf_early_data_cb(
```

Comment:
Can we re-use the mbedtls_ssl_tls13_conf_early_data to config the callback at once, just as the prototype does?